### ISSUE TITLE
Import Bloomsbury MARC records

### ISSUE THREAD

**Issue by mechanical-rus528** [internetarchive/openlibrary#9717]
### Problem

Import full MARC records from https://www.bloomsburycollections.com/marc-and-kbart-records. They seem to release an update every year. They'd probably be happy to see these get imported, though they don't explicitly specify a license. I could contact them to ask if this is okay to import, if you want.

**Comment by hornc** [comment]
This is currently in progress...

Example import: https://openlibrary.org/books/OL27833901M/Toy_Story?_compare=Compare&b=5&a=4&m=diff

This is from the Bloomsbury Open Access collection and the DOI (which is the persistent link to the open material) wasn't auto-imported.

It appears in the [MARC](https://openlibrary.org/show-records/marc_bloomsbury/BloomsburyCollections_ALL_BloomsburyOpenAccess.mrc:1112076:3334) as an `856` Electronic Location and Access HTTP URL

There are 526 Open Access records in the source MARC, I'll extract the DOIs and write them back to the imported / matched records.

**Comment by cdrini** [comment]
Oh that is awesome, thank you @hornc !!

**Comment by hornc** [comment]
This is a manual addition of the doi in the existing doi identifier field: https://openlibrary.org/books/OL27833901M/Toy_Story?_compare=Compare&b=6&a=5&m=diff

I'll figure out how to use the new `providers` field to put the URL in too:

```
"providers": [
  {
    "url": "https://doi.org/10.5040/9781501324949?locatt=label:secondary_bloomsburyCollections",
    "access": "read",
    "format": "web"
  }
]
```

There is a pdf download link from that page, but I'm not sure there is a good / persistent way to link to it, should `format` be `web` or `pdf` for this case?

**Comment by cdrini** [comment]
Yeah this is a tricky case... What OPDS calls an "indirectly" provided book, where the format is different than the thing that you'll see when you click. I think setting it to just web for now might make the most sense 👍

**Comment by hornc** [comment]
Import of Bloomsbury MARC records complete! Public access links is via the new DOI identifier/links I added in #10652. 
Thanks for the link to the data @mechanical-rus528 !
